refactor: maintain dracut version in dracut.sh

bdrung opened the pull request:
## Changes

The git release tag does not contain the `dracut-version.sh` file and therefore `git archive` will lack this file as well. The Makefile attempts to populate version information from git tags (which is impossible in an unpacked release tarball) and, failing that, reverts to whatever would be encoded in that file if it were sourced.

Maintain the dracut version directly in `dracut.sh` and call `dracut.sh --version` to get the version number. The version can be changed with a simple sed command to a new version (for releases or for distributors that want to include the packaging version):

```
sed -i "s/^DRACUT_VERSION=.*$/DRACUT_VERSION=\"${NEW_VERSION}\"/" dracut.sh
```

## Checklist
- [x] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #2169

aafeijoo-suse reviewed (commented):
I didn't want to mean this in https://github.com/dracutdevs/dracut/pull/2450#discussion_r1303857075, I was thinking of just setting `DRACUT_VERSION` at the beginning of `dracut.sh` and leave the rest of the code untouched.

If we have to `sed` any `.sh` script, I prefer #2450  and do it on the dedicated `dracut-version.sh`.

ahesford commented:
I don't have strong opinions on how to handle versioning in project repositories. It's a hard enough problem that, in Python, a million different packages exist to try to divine the version from various markers.

That said, the approach in this PR is pretty much how we handle versioning for ZFSBootMenu. A [releng/tag-release.sh](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/tag-release.sh) script does some git magic to extract release notes, prepare the branch for tagging, and [overwrites the version](https://github.com/zbm-dev/zfsbootmenu/blob/master/releng/tag-release.sh#L115) of the one script that requires it. It works well enough for us!

bdrung committed:
refactor: maintain dracut version in dracut.sh

Maintain the dracut version directly in `dracut.sh` and call
`dracut.sh --version` to get the version number. The version can be
changed with a simple sed command to a new version (for releases or for
distributors that want to include the packaging version):

```
sed -i "s/^DRACUT_VERSION=.*$/DRACUT_VERSION=\"${NEW_VERSION}\"/" dracut.sh
```

Fixes #2169
Signed-off-by: Benjamin Drung <benjamin.drung@canonical.com>

bdrung commented:
Changed `fix` to `refactor`.
